diff --git a/include/bx/file.h b/include/bx/file.h
index 88def0e..c851502 100644
--- a/include/bx/file.h
+++ b/include/bx/file.h
@@ -9,6 +9,14 @@
 #include "filepath.h"
 #include "readerwriter.h"
 
+#if defined(_LARGEFILE64_SOURCE)
+    /* for glibc system that support LFS */
+#else
+    /* others */
+#   define fseeko64 fseeko
+#   define ftello64 ftello
+#endif
+
 namespace bx
 {
 	/// Returns standard input reader.
